File: src/main/scala/Problem77.scala
package scala

import utils.Euler
import scala.jdk.CollectionConverters.CollectionHasAsScala

object Problem77 extends App {
  val p = Array.fill(1000){0}
  p(0) = 1
  val primes = Euler.primeList(1000).asScala.map(_.toInt)
  for (n <- primes; i <- n until p.length)
    p(i) += p(i-n)
  println(p.zipWithIndex.find(_._1 > 5000).get._2)
}

File: src/main/java/Problem77.java
import java.util.List;

import utils.Euler;

public class Problem77 {
	public static void main(String[] args) {
		System.out.println(solution());
	}

	public static int solution() {
		for (int tallet = 0; tallet <= 100; tallet++) {
			
			List<Integer> list = Euler.primeList(tallet);
			int [] array = new int[tallet+1]; 
			for (int i = 0; i < list.size(); i++) {
				int tall = list.get(i);
				++array[tall];
				for (int j = tall; j < array.length; ++j) {
					array[j]+=array[j-tall];
				}
			}
			if (array[tallet]>5000) {
				return tallet;
			}
//				System.out.println(array[tallet]);
		}
		return 0;
	}
}
